app/src/main/java/org/simple/clinic/bp/BloodPressureMeasurement.kt

```diff
@@ -77,6 +77,13 @@ data class BloodPressureMeasurement (
     @Query("SELECT COUNT(uuid) FROM bloodpressuremeasurement")
     fun count(): Flowable<Int>
 
+    @Query("""
+      SELECT COUNT(uuid)
+      FROM bloodpressuremeasurement
+      WHERE patientUuid = :patientUuid AND deletedAt IS NULL
+    """)
+    fun recordedBloodPressureCountForPatient(patientUuid: UUID): Flowable<Int>
```

Comment:
When we "delete" a BP, we are basically indicating that the BP does not exist and it's an implementation detail that for us, deleting a BP means setting a property (`deletedAt`) on a record. This implementation detail should not be exposed externally in any way. For instance, we wouldn't be talking about renaming this method if instead of setting a property, we were actually deleting the record.

The assumption is that all data access methods will exclude deleted records unless and until we specifically need to access deleted records (say, for sync) and these methods should be named appropriately. This is what we have done so far for all our models.

Reply:
Got it!